### PR TITLE
Do not store mtp_losses/mtp_acceptance in state params

### PR DESCRIPTION
**Pull request by bvandermoon**
# Description

* Do not store `mtp_losses` and `mtp_acceptance` as part of params. This was causing a checkpoint loading error because they are not stored as part of the checkpoints.
* Update self attribute names to match pre-NNX checkpoints
* NOTE: Looks like pyink wants to update the format for the whole file. I'll do this in a follow-up PR to avoid losing the diff of this change

FIXES: b/483723849

# Tests

Successfully ran the XPK workload in the linked bug with checkpoint loading.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/MaxText/layers/multi_token_prediction.py [hunk 1]
```diff
@@ -199,6 +199,8 @@ def __init__(
     self.mtp_preds = mtp_acceptance(jnp.zeros((1,), dtype=jnp.float32))
     self.mtp_mask = mtp_acceptance(jnp.zeros((1,), dtype=jnp.float32))
 
+    nnx.pop(self, (mtp_losses, mtp_acceptance))
```
Comment (cgarciae) [inline src/MaxText/layers/multi_token_prediction.py]: Its a bit weird to `pop` on `__init__`, specially because you just created some Variables above, these would be lost.

File: src/MaxText/layers/multi_token_prediction.py
Comment (cgarciae) [inline src/MaxText/layers/multi_token_prediction.py]: isn't it a bit easier just to delete these lines instead of assigning them and then using `pop`?

Comment (bvandermoon) [reply src/MaxText/layers/multi_token_prediction.py]: Yes good point, thanks @cgarciae

File: src/MaxText/layers/multi_token_prediction.py (hunk 1 above)
Comment (bvandermoon) [reply src/MaxText/layers/multi_token_prediction.py]: Yes good point, thanks @cgarciae. Updated the PR
